cmd/nknc/commands/id.go

```diff
@@ -0,0 +1,137 @@
+package commands
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"os"
+
+	api "github.com/nknorg/nkn/v2/api/common"
+	"github.com/nknorg/nkn/v2/api/httpjson/client"
+	"github.com/nknorg/nkn/v2/common"
+	"github.com/nknorg/nkn/v2/config"
+	"github.com/nknorg/nkn/v2/util/log"
+	"github.com/nknorg/nkn/v2/vault"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// idCmd represents the id command
+var idCmd = &cobra.Command{
+	Use:   "id",
+	Short: "generate id for nknd",
+	Long:  "With nknc id, you could generate ID.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		showusage := true
+		cmd.Flags().Visit(func(name *pflag.Flag) {
+			showusage = false
+		})
+		if showusage {
+			return cmd.Usage()
+		}
+		if err := generateIDAction(); err != nil {
+			log.Error(err)
+		}
+		return nil
+	},
+}
+
+var (
+	genid  bool
+	pubkey string
+)
+
+func init() {
+	rootCmd.AddCommand(idCmd)
+
+	idCmd.Flags().BoolVar(&genid, "genid", false, "generate id")
+	idCmd.Flags().StringVar(&pubkey, "pubkey", "", "pubkey to generate id for, leave empty for local wallet pubkey")
+	idCmd.Flags().StringVarP(&WalletFile, "wallet", "w", config.Parameters.WalletFile, "wallet name")
+	idCmd.Flags().StringVarP(&WalletPassword, "password", "p", "", "wallet password")
+	idCmd.Flags().StringVar(&regfee, "regfee", "", "registration fee")
+	idCmd.Flags().StringVarP(&fee, "fee", "f", "", "transaction fee")
+	idCmd.Flags().Uint64Var(&Nonce, "nonce", 0, "nonce")
+}
+
+func generateIDAction() error {
+	pubkeyHex := pubkey
+	pubkey, err := hex.DecodeString(pubkeyHex)
```

Comment:
A little confusion for readability.
`pubkey in L57` is a global var type string.
`pubkey in L58` is a local var type byte slice.
Local var override global var at here.